### PR TITLE
Add plugin: Custom Note Width

### PR DESCRIPTION
**Pull request by 0skater0**
# I am submitting a new Community Plugin

## Repo URL
https://github.com/0skater0/obsidian-custom-note-width

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by joethei** [comment]
[export function debounce<T extends any[]>(func: (...args: T) => void, wait: number): (...args: T) => void](https://github.com/0skater0/obsidian-custom-note-width/blob/7c482075435f1240bc9ba76235fcc15b3355a636/src/utility/utilities.ts#L47-L58)
The Obsidian API already includes this function.

[export const OBSIDIAN_DIR = ".obsidian";](https://github.com/0skater0/obsidian-custom-note-width/blob/7c482075435f1240bc9ba76235fcc15b3355a636/src/utility/constants.ts#L19)
This can be changed in the settings, to get the path to the plugin folder use `this.manifest.dir` from the plugin instance.
You can also get the plugin name from that.,

[console.error("Slider-Element konnte nicht gefunden werden.");](https://github.com/0skater0/obsidian-custom-note-width/blob/7c482075435f1240bc9ba76235fcc15b3355a636/src/ui/uiManager.ts#L47)
Please use english for error messages.

[const content = await this.app.vault.read(file);](https://github.com/0skater0/obsidian-custom-note-width/blob/7c482075435f1240bc9ba76235fcc15b3355a636/src/note/yamlFrontMatterProcessor.ts#L38-L66)
Use the metadata API for this: <https://github.com/obsidianmd/obsidian-api/blob/master/CHANGELOG.md#new-metadata-api>

[updateTimeout: NodeJS.Timeout;](https://github.com/0skater0/obsidian-custom-note-width/blob/7c482075435f1240bc9ba76235fcc15b3355a636/src/event/eventHandler.ts#L12)
Use a regular number type here, and use the `window.setTimeout` function below instead.

**Comment by 0skater0** [comment]
> [export function debounce<T extends any[]>(func: (...args: T) => void, wait: number): (...args: T) => void](https://github.com/0skater0/obsidian-custom-note-width/blob/7c482075435f1240bc9ba76235fcc15b3355a636/src/utility/utilities.ts#L47-L58) The Obsidian API already includes this function.
> 
> [export const OBSIDIAN_DIR = ".obsidian";](https://github.com/0skater0/obsidian-custom-note-width/blob/7c482075435f1240bc9ba76235fcc15b3355a636/src/utility/constants.ts#L19) This can be changed in the settings, to get the path to the plugin folder use `this.manifest.dir` from the plugin instance. You can also get the plugin name from that.,
> 
> [console.error("Slider-Element konnte nicht gefunden werden.");](https://github.com/0skater0/obsidian-custom-note-width/blob/7c482075435f1240bc9ba76235fcc15b3355a636/src/ui/uiManager.ts#L47) Please use english for error messages.
> 
> [const content = await this.app.vault.read(file);](https://github.com/0skater0/obsidian-custom-note-width/blob/7c482075435f1240bc9ba76235fcc15b3355a636/src/note/yamlFrontMatterProcessor.ts#L38-L66) Use the metadata API for this: https://github.com/obsidianmd/obsidian-api/blob/master/CHANGELOG.md#new-metadata-api
> 
> [updateTimeout: NodeJS.Timeout;](https://github.com/0skater0/obsidian-custom-note-width/blob/7c482075435f1240bc9ba76235fcc15b3355a636/src/event/eventHandler.ts#L12) Use a regular number type here, and use the `window.setTimeout` function below instead.

I have addressed this

**Comment by joethei** [comment]
What did you do to your old commits?

**Comment by 0skater0** [comment]
Somehow I pushed to an existing commit. I have fixed the repo and it should now show the commits.

> [export function debounce<T extends any[]>(func: (...args: T) => void, wait: number): (...args: T) => void](https://github.com/0skater0/obsidian-custom-note-width/blob/38430de49e4af3a3022cd9c976fde6333a1b0b9e/src/utility/utilities.ts#L47-L58) 
> The Obsidian API already includes this function.
> 
> [export const OBSIDIAN_DIR = ".obsidian";](https://github.com/0skater0/obsidian-custom-note-width/blob/38430de49e4af3a3022cd9c976fde6333a1b0b9e/src/utility/constants.ts#L19) 
> This can be changed in the settings, to get the path to the plugin folder use `this.manifest.dir` from the plugin instance. You can also get the plugin name from that.,
> 
> [console.error("Slider-Element konnte nicht gefunden werden.");](https://github.com/0skater0/obsidian-custom-note-width/blob/38430de49e4af3a3022cd9c976fde6333a1b0b9e/src/ui/uiManager.ts#L47) 
> Please use english for error messages.
> 
> [const content = await this.app.vault.read(file);](https://github.com/0skater0/obsidian-custom-note-width/blob/38430de49e4af3a3022cd9c976fde6333a1b0b9e/src/note/yamlFrontMatterProcessor.ts#L38-L66) 
> Use the metadata API for this: https://github.com/obsidianmd/obsidian-api/blob/master/CHANGELOG.md#new-metadata-api
> 
> [updateTimeout: NodeJS.Timeout;](https://github.com/0skater0/obsidian-custom-note-width/blob/38430de49e4af3a3022cd9c976fde6333a1b0b9e/src/event/eventHandler.ts#L12) 
> Use a regular number type here, and use the `window.setTimeout` function below instead.

**Comment by liamcain** [comment]
- [querySelector](https://github.com/0skater0/obsidian-custom-note-width/blob/97948332826310d1945c84ccef6931799e7b03c5/src/statusbar/statusBarManager.ts#L50) why are you querying the dom for this? You can just keep a reference to statusBarItem and do statusBarItem.detach()
- [previousWindowState](https://github.com/0skater0/obsidian-custom-note-width/blob/97948332826310d1945c84ccef6931799e7b03c5/src/event/eventHandler.ts#L28) Looks like this doesn't handle [pop-out windows](https://help.obsidian.md/User+interface/Pop-out+windows) at all.
- [// Override methods](https://github.com/0skater0/obsidian-custom-note-width/blob/97948332826310d1945c84ccef6931799e7b03c5/src/event/eventHandler.ts#L139) there's no need to monkeypatch these functions. Just listen for 'resize' <https://github.com/obsidianmd/obsidian-api/blob/249eb773347588f4e9fec7c600e1904cd9240fcb/obsidian.d.ts#L4546>

**Comment by 0skater0** [comment]
> * [querySelector](https://github.com/0skater0/obsidian-custom-note-width/blob/97948332826310d1945c84ccef6931799e7b03c5/src/statusbar/statusBarManager.ts#L50) why are you querying the dom for this? You can just keep a reference to statusBarItem and do statusBarItem.detach()
> 
>     * [previousWindowState](https://github.com/0skater0/obsidian-custom-note-width/blob/97948332826310d1945c84ccef6931799e7b03c5/src/event/eventHandler.ts#L28) Looks like this doesn't handle [pop-out windows](https://help.obsidian.md/User+interface/Pop-out+windows) at all.
> 
>     * [// Override methods](https://github.com/0skater0/obsidian-custom-note-width/blob/97948332826310d1945c84ccef6931799e7b03c5/src/event/eventHandler.ts#L139) there's no need to monkeypatch these functions. Just listen for 'resize' https://github.com/obsidianmd/obsidian-api/blob/249eb773347588f4e9fec7c600e1904cd9240fcb/obsidian.d.ts#L4546

I've updated the code.
